corehq/util/couch_doc_processor.py

```diff
@@ -367,3 +395,65 @@ def _processing_complete(self):
         ))
         if self.skipped:
             print(DOCS_SKIPPED_WARNING.format(self.skipped))
+
+
+class BulkDocProcessorEventHandler(PaginateViewEventHandler):
+
+    def __init__(self, processor):
+        self.processor = processor
```

Comment:
Not sure how important this is, but this would be a good place to use a [`weakref.ref`](https://docs.python.org/2/library/weakref.html#weakref.ref) to break the circular reference in the object graph.